gpflow/utilities/utilities.py

```diff
@@ -382,12 +383,9 @@ def freeze(input_module: M) -> M:
     :param input_module: tf.Module or gpflow.Module.
     :return: Returns a frozen deepcopy of an input object.
     """
-    module_copy = deepcopy(input_module)
-    for name, value in parameter_dict(module_copy).items():
-        tensor = tf.convert_to_tensor(value, dtype=value.dtype)
-        const_value = tf.constant(tensor)
-        attr_path = name.lstrip(".")
-        setattr_by_path(module_copy, attr_path, const_value)
+    objects_to_freeze = _get_leaf_components(input_module)
+    memo_tensors = {id(v): tf.constant(tf.convert_to_tensor(v)) for v in objects_to_freeze.values()}
```

Comment:
```suggestion
    memo_tensors = {id(v): tf.convert_to_tensor(v) for v in objects_to_freeze.values()}
```
the tf.constant() doesn't do anything:
```python
a = tf.convert_to_tensor(1.0)
b = tf.constant(a)
assert a is b
```